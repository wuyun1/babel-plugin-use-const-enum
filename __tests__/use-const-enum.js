import { transformAsync } from '@babel/core';
import plugin from '../src';



it('Transforms no initializers', async () => {
  const input = `import { FieldType } from 'choerodon-ui/pro/lib/data-set/enum';

  enum Status {
    pending = 'pending',
    end = 'end',
    start = 'start'
  }

  console.log(FieldType.string);
  console.log(FieldType["string-aaa"]);
  console.log(Status.end);
`;

  const options = {
    plugins: [[plugin, { "importRegStr": "choerodon-ui\/.*\/enum" }]],
  };

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();
});
