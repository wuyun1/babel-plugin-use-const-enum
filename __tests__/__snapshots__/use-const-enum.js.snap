// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transforms no initializers 1`] = `
"import { FieldType } from 'choerodon-ui/pro/lib/data-set/enum';
enum Status {
  pending = 'pending',
  end = 'end',
  start = 'start',
}
console.log(\\"string\\");
console.log(Status.end);"
`;
